File: src/main/java/com/adri/pfm/pfm_user_service/controller/UserController.java
package com.adri.pfm.pfm_user_service.controller;

import com.adri.pfm.pfm_user_service.dto.CreateUserAccountDTO;
import com.adri.pfm.pfm_user_service.dto.UserAccountDTO;
import com.adri.pfm.pfm_user_service.service.UserAccountService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/v1/user-accounts")
public class UserController {

    private final UserAccountService userAccountService;

    @PostMapping
    public ResponseEntity<UserAccountDTO> createCategory(@RequestBody CreateUserAccountDTO createUserAccountDTO) {
        return ResponseEntity.ok(userAccountService.createUserAccount(createUserAccountDTO));
    }

    @GetMapping("/{userAccountId}")
    public ResponseEntity<UserAccountDTO> findById(@PathVariable long userAccountId) {
        return ResponseEntity.ok(userAccountService.findByUserAccountId(userAccountId));
    }

    @GetMapping("/username/{username}")
    public ResponseEntity<UserAccountDTO> findByUsername(@PathVariable String username) {
        return ResponseEntity.ok(userAccountService.findByUserAccountUsername(username));
    }
}

File: src/main/kotlin/com/amf/pfm/user/controller/UserController.kt
package com.amf.pfm.user.controller

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/users")
class UserController {

    data class User(
        val id: UUID = UUID.randomUUID(),
        val name: String,
        val username: String,
        val password: String
    )

    companion object {
        private val users = mutableMapOf<UUID, User>()
    }

    @PostMapping
    fun createUser(@RequestBody userRequest: CreateUserRequest): ResponseEntity<User> {
        val user = User(
            name = userRequest.name,
            username = userRequest.username,
            password = userRequest.password
        )
        users[user.id] = user
        return ResponseEntity.ok(user)
    }

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: UUID): ResponseEntity<User> {
        val user = users[id]
        return if (user != null) ResponseEntity.ok(user)
        else ResponseEntity.notFound().build()
    }

    @GetMapping
    fun getAllUsers(): ResponseEntity<List<User>> {
        return ResponseEntity.ok(users.values.toList())
    }

    data class CreateUserRequest(
        val name: String,
        val username: String,
        val password: String
    )
}